### PR TITLE
refact: better dynamic constructors

### PR DESCRIPTION
**Pull request by antmat**
@3hren PTAL

### CODE REVIEW

File: include/cocaine/dynamic/constructors.hpp [hunk 1]
```diff
@@ -330,68 +330,59 @@ struct dynamic_constructor<std::map<std::string, dynamic_t>> {
...
-    convert(const std::map<std::string, T>& from, dynamic_t::value_t& to) {
-        dynamic_constructor<dynamic_t::object_t>::convert(dynamic_t::object_t(), to);
-
-        auto& object = boost::get<detail::dynamic::incomplete_wrapper<dynamic_t::object_t>>(to).get();
-
-        for(auto it = from.begin(); it != from.end(); ++it) {
-            object.insert(dynamic_t::object_t::value_type(it->first, it->second));
-        }
+    std::string
+    cast(const std::string& from) {
```
Comment (3Hren) [inline include/cocaine/dynamic/constructors.hpp]: Lexical cast is already no-op for strings.

File: include/cocaine/dynamic/constructors.hpp [hunk 2]
```diff
@@ -330,68 +330,59 @@ struct dynamic_constructor<std::map<std::string, dynamic_t>> {
...
+        return boost::lexical_cast<std::string>(std::forward<From>(from));
     }
-};
 
-template<class T>
-struct dynamic_constructor<std::unordered_map<std::string, T>> {
-    static const bool enable = true;
+    // helper to choose proper implementaion
+    template<class From>
+    struct is_simple {
```
Comment (3Hren) [inline include/cocaine/dynamic/constructors.hpp]: Use template alias here.

Comment (3Hren) [inline include/cocaine/dynamic/constructors.hpp]: And rename to something meaningful.

Comment (antmat) [inline include/cocaine/dynamic/constructors.hpp]: sorry, don't get it

Comment (antmat) [inline include/cocaine/dynamic/constructors.hpp]: it's template with static bool member - how can I alias it?

Comment (3Hren) [inline include/cocaine/dynamic/constructors.hpp]: template<class From>
using govno = std::is_same<...>;

**Pull request merged**